frontend/src/App.js

```diff
@@ -61,177 +62,198 @@ const NotFound = () => (
 )
 
 const App = (props) => {
+  const {
+    updateIsLoading,
+    loginUser,
+    fetchRegistrationManagement,
+    setError,
+    initializeMyGroup,
+    logoutUser,
+    clearRegistrations,
+    isLoading,
+    user,
+  } = props
+
   useEffect(() => {
-    props.updateIsLoading(true)
-    if (!window.location.href.includes('customer-review/')){
-      props.loginUser()
-    }
-    fetchRegistrationManagement()
-    props.updateIsLoading(false)
-    setInterval(() => {
-      console.log('timeout')
-      if (!window.location.href.includes('customer-review/')){
-        loginService.login()
-      }
+    const handleLogin = () =>
+      !window.location.href.includes('customer-review/') && loginUser()
+
+    // TODO: these should prolly use async/await?
```

Comment:
Figure this out

Reply:
Not sure, but counter question is if this should be addressed in this branch? App has been differently defined in current main-branch.

Reply:
I'll do this as I had to make some changes there!

Reply:
It was incorrectly done: it set the loading status `updateIsLoading(true)` and then fetched a bunch of stuff and set the loading status again `updateIsLoading(false)`. But it didn't use async/await so it wouldn't wait for the data fetching and therefore the loading status was immediately set to false, before the data was fetched.